src/main/kotlin/it/unibo/protelis/web/execution/simulated/NewSimulatedProtelisEngine.kt

```diff
@@ -0,0 +1,82 @@
+package it.unibo.protelis.web.execution.simulated
+
+import it.unibo.alchemist.core.implementations.Engine
+import it.unibo.alchemist.core.interfaces.Simulation
+import it.unibo.alchemist.core.interfaces.Status
+import it.unibo.alchemist.core.interfaces.waitForAndCheck
+import it.unibo.alchemist.loader.YamlLoader
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
+import it.unibo.alchemist.model.implementations.times.DoubleTime
+import it.unibo.protelis.web.execution.ProtelisObserver
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class NewSimulatedProtelisEngine {
+  private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+  private var alchemistEngine: Simulation<Any, Euclidean2DPosition>? = null
+
+  fun setup(sourceCode: String, monitor: ProtelisObserver) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [UndocumentedPublicFunction - [setup] at /src/src/main/kotlin/it/unibo/protelis/web/execution/simulated/NewSimulatedProtelisEngine.kt:20:3](https://app.codacy.com/manual/NiccoMlt/Protelis-Web/pullRequest?prid=5040950)